app/service/rest.py

```diff
@@ -945,10 +948,35 @@ def check_if_reply_to_address_already_in_use(service_id, email_address):
 
 @service_blueprint.route('/<uuid:service_id>/returned-letter-summary', methods=['GET'])
 def returned_letter_summary(service_id):
-    results = get_returned_letter_summary(service_id)
+    results = fetch_returned_letter_summary(service_id)
 
     json_results = [{'returned_letter_count': x.returned_letter_count,
                      'reported_at': x.reported_at.strftime(DATE_FORMAT)
                      } for x in results]
 
     return jsonify(json_results)
+
+
+@service_blueprint.route('/<uuid:service_id>/returned-letters', methods=['GET'])
+def get_returned_letters(service_id):
+    results = fetch_returned_letters(service_id=service_id, report_date=request.args.get('reported_at'))
+
+    json_results = [
+        {'notification_id': x.notification_id,
+         # client reference can only be added on API letters
+         'client_reference': x.client_reference if x.api_key_id else None,
+         'reported_at': x.reported_at.strftime(DATE_FORMAT),
+         'created_at': x.created_at.strftime("%Y-%m-%d %H:%M:%S"),
+         # it doesn't make sense to show hidden/precompiled templates
+         'template_name': x.template_name if not x.hidden else None,
+         'template_id': x.template_id if not x.hidden else None,
+         'template_version': x.template_version if not x.hidden else None,
+         'user_name': x.user_name or 'API',
+         'email_address': x.email_address or 'API',
+         'original_file_name': x.original_file_name,
+         'job_row_number': x.job_row_number,
+         # the file name for a letter uploaded via the UI
+         'uploaded_letter': x.client_reference if x.hidden and not x.api_key_id else None
```

Comment:
Could the name of this key be slightly more descriptive without having to come look back here at the comment please? Maybe `uploaded_letter_file_name`?